driver/src/main/java/org/neo4j/driver/internal/NetworkSession.java

```diff
@@ -225,6 +247,47 @@ public synchronized void onConnectionError( boolean recoverable )
         }
     }
 
+    private synchronized <T> T transaction( AccessMode mode, Function<Transaction,T> work )
+    {
+        RetryDecision decision = null;
+        List<Throwable> errors = null;
+
+        while ( true )
+        {
+            try ( Transaction tx = beginTransaction( mode ) )
+            {
+                return work.apply( tx );
+            }
+            catch ( Throwable newError )
+            {
+                decision = retryLogic.apply( newError, decision );
+
+                if ( decision.shouldRetry() )
+                {
+                    errors = recordError( newError, errors );
```

Comment:
Can we just directly chain the errors? why we have to first remember them (in `recordError`) and then build the error chain (in `addSuppressed`)?